### ISSUE TITLE
BatchNorm layer 'num_batches_tracked' overwritten with default value when loading empty state_dict

### ISSUE THREAD

**Issue by slin96** [pytorch/pytorch#110361]
### 🐛 Describe the bug

BatchNorm layer 'num_batches_tracked' overwritten with default value when loading empty state_dict.

Probably should be fixed in: /torch/nn/modules/batchnorm.py lines 103-108 by checking if a value already exists.

```python
from collections import OrderedDict
from torchvision.models import mobilenet_v2, MobileNet_V2_Weights

sd_key = 'features.0.1.num_batches_tracked'
model = mobilenet_v2(weights=MobileNet_V2_Weights.IMAGENET1K_V1.DEFAULT)
print(f"before loading: {model.state_dict()['features.0.1.num_batches_tracked']}")

empty_dict = OrderedDict()
model.load_state_dict(empty_dict, strict=False)
print(f"after loading: {model.state_dict()['features.0.1.num_batches_tracked']}")
```

expected output: 
```
before loading: 736839
after loading: 736839
```

actual output: 
```
before loading: 736839
after loading: 0
```

### Versions

Collecting environment information...
PyTorch version: 2.0.1
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 13.4 (x86_64)
GCC version: Could not collect
Clang version: 14.0.3 (clang-1403.0.22.14.1)
CMake version: Could not collect
Libc version: N/A

Python version: 3.9.10 (main, Jan 15 2022, 11:48:04)  [Clang 13.0.0 (clang-1300.0.29.3)] (64-bit runtime)
Python platform: macOS-13.4-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M1 Pro

Versions of relevant libraries:
[pip3] numpy==1.25.1
[pip3] torch==2.0.1
[pip3] torchinfo==1.8.0
[pip3] torchvision==0.15.2
[pip3] torchviz==0.0.2

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki

**Comment by slin96** [comment]
Changing line 108 to: `state_dict[num_batches_tracked_key] = self.num_batches_tracked if self.num_batches_tracked is None else torch.tensor(0, dtype=torch.long)` solves it for me, don't know if that has any unwanted effects in other scenarios though.

**Comment by slin96** [comment]
Changing line 108 in /torch/nn/modules/batchnorm.py to: `state_dict[num_batches_tracked_key] = self.num_batches_tracked if self.num_batches_tracked is not None else torch.tensor(0, dtype=torch.long)` solves it for me, don't know if that has any unwanted effects in other scenarios though.
